client/my-sites/comments/comment-list/index.jsx

```diff
@@ -132,18 +146,30 @@ export class CommentList extends Component {
 		} );
 	};
 
-	setCommentStatus = ( commentId, postId, status, options = { showNotice: true } ) => {
+	setCommentStatus = ( commentId, postId, status, options = { isUndo: false, persist: false, showNotice: true } ) => {
+		const { isUndo, persist, showNotice } = options;
+
 		// TODO: Replace with Redux getComment()
 		const comment = this.getComment( commentId );
 
 		if ( comment && status === comment.status ) {
 			return;
 		}
 
+		if ( persist ) {
+			this.updatePersistedComments( {
+				...comment,
+				i_like: 'approved' !== status ? false : comment.i_like,
+				status,
+			}, isUndo );
```

Comment:
Ok this is the problem here: it's true that the data is available in the Redux state, but not in the component.
The `getSiteComments` selector takes care of filtering the comments list by status, so that `CommentList` only knows about the comments belonging to the currently displayed list.
In other words: as soon as I change a comment status, that comment disappears from the portion of state passed to `CommentList`.

In order to persist it (and make it work seamlessly), I both have to store it _and_ manually update its `status` and `i_like` values (according to the status change action I performed).
This is unavoidable for the same reason as above: I can't copy the comment _after_ the status change because it wouldn't be there anymore.

One way to change this would be that `getSiteComments` returns all comments (or, eventually, comment IDs), and delegates to `CommentList.render()` the burden of filtering by status.
This way we'd have access to everything, but at the same time we'd make `CommentList` aware of (a potentially huge amount of) stuff it doesn't (and might never) really care.